### PR TITLE
Add ScaleD functional Option

### PR DESCRIPTION
**Pull request by Jinp800125**
[Changes]
    1. TensileLite
    2. hipBLASLt
    3. tunning library
    4. gTest
    5. localTest
    6. example

### CODE REVIEW

File: clients/gtest/matmul_gtest.yaml [hunk 1]
```diff
@@ -498,6 +498,18 @@ Tests:
...
+  category: pre_checkin
+  function:
+    matmul: *real_precisions
+  M: [128, 129]
+  N: [128, 129]
+  K: [128, 129]
+  transA_transB: *transA_transB_range
+  alpha: 1
+  beta: [ 0.0, 2.0 ]
+  scaleD_vector: [0, 1]
```
Comment (jichangjichang) [inline clients/gtest/matmul_gtest.yaml:512]: non-scaleD case has listed in other test.

File: library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/Equality/aldebaran_Cijk_Alik_Bljk_BBS_BH_Bias_AH.yaml [hunk 2]
```diff
@@ -223,13 +224,14 @@
       TransposeB: 0
       UseBeta: true
       UseBias: true
+      UseScaleD: true
       UseInitialStridesAB: false
       UseInitialStridesCD: false
     ScheduleGlobalRead: 1
     ScheduleIterAlg: 3
     ScheduleLocalWrite: 1
     SolutionIndex: 0
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/Equality/aldebaran_Cijk_Alik_Bljk_BBS_BH_Bias_AH.yaml:233]: file name of this yaml file should be revised accordingly

File: tensilelite/Tensile/KernelWriterAssembly.py [hunk 3]
```diff
@@ -3212,7 +3212,7 @@ def endSummation(self, kernel, label = None):
     storeSgprLoad = 0
     runActivation = True if ((kernel["ProblemType"]["ActivationType"] != 'none') and (kernel["GlobalSplitU"] == 1) \
         and kernel["ActivationFused"]) else False
-    if kernel["ProblemType"]["UseBias"] and (kernel["GlobalSplitU"] == 1):
+    if kernel["ProblemType"]["UseBias"] and (kernel["GlobalSplitU"] == 1):# ScaleD might reference
```
Comment (jichangjichang) [inline tensilelite/Tensile/KernelWriterAssembly.py:3215]: old note?

File: tensilelite/Tensile/KernelWriterAssembly.py [hunk 4]
```diff
@@ -5516,6 +5516,11 @@ def globalWriteWorkGroupInit(self, kernel):
     if kernel["BufferStore"]:
       module.add(self.allocPostLoopSrd("D"))
       module.add(self.allocPostLoopSrd("C"))
+      if kernel["ProblemType"]["UseScaleD"] and (kernel["GlobalSplitU"] == 1):
+        labelStr = self.labels.getNameInc("ScaleD")
+        module.add(allocPostLoopSrdSuppress("ScaleD", labelStr))
+        ScaleBpe = int(4)# size of SrdScaleD scaled by BPE
```
Comment (jichangjichang) [inline tensilelite/Tensile/KernelWriterAssembly.py:5522]: don't use hard code.

File: clients/samples/example_hipblaslt_preference.cpp [hunk 5]
```diff
@@ -1010,7 +1049,7 @@ int main(int argc, char* argv[])
     if(header)
     {
         std::cout << "transAB, M, N, K, lda, ldb, ldc, stride_a, stride_b, "
-                     "stride_c, batch_count, alpha, beta, bias, activationType";
+                     "stride_c, batch_count, alpha, beta, bias, ,scaleD, activationType";
```
Comment (jichangjichang) [inline clients/samples/example_hipblaslt_preference.cpp:1052]: typo

**Review by jichangjichang — COMMENTED** [review]
Please check getKernelName() of KernelWriterConversion.py and
outputConversionKernelName() of ContractionSolution.cpp.
The kernel naming should be revised for ScaleD enabled case.

File: clients/samples/example_hipblaslt_preference.cpp [hunk 6]
```diff
@@ -207,6 +208,7 @@ void mat_mul_bias_activation(Tc             alpha,
                     + (bias == nullptr ? 0 : bias[i1]);
                 if(actType != ActivationType::NONE)
                     t = actFunc(t);
+                t                                    = (scaleD == nullptr ? 1.0 : scaleD[i1]);
```
Comment (jichangjichang) [inline clients/samples/example_hipblaslt_preference.cpp:211]: .......
Did you ever verify it?

**Review by jichangjichang — CHANGES_REQUESTED** [review]
please check comment

File: clients/samples/example_hipblaslt_preference.cpp (hunk 6 above)
Comment (Jinp800125) [reply clients/samples/example_hipblaslt_preference.cpp]: It weird...
Just testing at started moment.

**Review by jichangjichang — APPROVED** [review]

File: clients/include/testing_matmul.hpp [hunk 7]
```diff
@@ -58,29 +60,60 @@ void epilogue_func(int64_t m,
...
-        Ti bias_data = enable_bias ? static_cast<Ti>(*(bias + i)) : 0;
+        Ti bias_data   = enable_bias ? static_cast<Ti>(*(bias + i)) : 0;
+        Ti scaleD_data = enable_scaleD ? static_cast<Ti>(*(scaleD + i)) : 1;
 #pragma omp parallel for
         for(int j = 0; j < n; j++)
         {
             auto pos     = j * ld + i;
             auto in_Tact = static_cast<Tact>(*(in + pos)) + bias_data;
+            in_Tact *= scaleD_data;
             *(out + pos) = saturate_o(act_func(in_Tact, arg1, arg2));
```
Comment (jichangjichang) [inline clients/include/testing_matmul.hpp:71]: Does scaleD happen before activation? or after activation??

File: clients/include/testing_matmul.hpp [hunk 8]
```diff
@@ -500,7 +563,30 @@ void testing_matmul(const Arguments& arg)
                     }
                 }
             }
+            else if(arg.scaleD_vector)
```
Comment (jichangjichang) [inline clients/include/testing_matmul.hpp:566]: This code section seems redundant. epilogue_on case should able to cover it.

**Pull request merged**